src/Common/tests/Collections/IEnumerableTest.cs

```diff
@@ -0,0 +1,428 @@
+                if (IsResetNotSupported)
+                    enumerator = enumerable.GetEnumerator();
+                else
+                    enumerator.Reset();
+            }
+        }
+
+        private void RepeatTest(Action<IEnumerator, object[]> testCode, int iters = 3)
+        {
+            RepeatTest((e, i, it) => testCode(e, i), iters);
+        }
+
+        [Fact]
+        public void MoveNextHitsAllItems()
+        {
+            RepeatTest((enumerator, items) =>
+                       {
+                           var iterations = 0;
+                           while (enumerator.MoveNext())
+                           {
+                               iterations++;
+                           }
+                           Assert.Equal(items.Length, iterations);
+                       });
+        }
+
+        [Fact]
+        public void CurrentThrowsAfterEndOfCollection()
+        {
+            if (IsGenericCompatibility)
+                return; // apparently it is okay if enumerator.Current doesn't throw when the collection is generic.
+
+            RepeatTest((enumerator, items) =>
+                       {
+                           while (enumerator.MoveNext())
+                           {
+                           }
+                           Assert.Throws<InvalidOperationException>(
+                               () => enumerator.Current);
+                       });
+        }
+
+        [Fact]
+        public void MoveNextFalseAfterEndOfCollection()
+        {
+            RepeatTest((enumerator, items) =>
+                       {
+                           while (enumerator.MoveNext())
+                           {
+                           }
+
+                           Assert.False(enumerator.MoveNext());
+                       });
+        }
+
+        [Fact]
+        public void Current()
+        {
+            RepeatTest((enumerator, items, iteration) =>
+                       {
+                           if (iteration == 1)
+                               VerifyEnumerator(items, enumerator, 0, items.Length / 2, false, true);
+                           else
+                               VerifyEnumerator(items, enumerator);
+                       });
+        }
+
+        [Fact]
+        public void Reset()
+        {
+            if (IsResetNotSupported)
+            {
+                RepeatTest((enumerator, items) =>
+                           {
+                               Assert.Throws<NotSupportedException>(
+                                   () => enumerator.Reset());
+                           });
+                RepeatTest((enumerator, items, iter) =>
+                           {
+                               if (iter == 1)
+                               {
+                                   VerifyEnumerator(items, enumerator, 0, items.Length/2, false, true);
+                                   for (int i = 0; i < 3; i++)
+                                   {
```

Comment:
Its what the original test did. I'ts probably not necessary but I left it to be consistent.